src/Gui/NotificationBox.cpp

```diff
@@ -0,0 +1,352 @@
+}
+void NotificationLabel::restartExpireTimer(int msecDisplayTime)
+{
+    int time = 10000 + 40 * qMax(0, text().length()-100);
+    if (msecDisplayTime > 0) {
+        time = msecDisplayTime;
+    }
+    expireTimer.start(time);
+    hideTimer.stop();
+}
+void NotificationLabel::reuseNotification(const QString &text, int msecDisplayTime, const QPoint &pos)
+{
+    setText(text);
+    updateSize(pos);
+    restartExpireTimer(msecDisplayTime);
+}
+void NotificationLabel::updateSize(const QPoint &pos)
+{
+    // Ensure that we get correct sizeHints by placing this window on the right screen.
+    QFontMetrics fm(font());
+    QSize extra(1, 0);
+    // Make it look good with the default ToolTip font on Mac, which has a small descent.
+    if (fm.descent() == 2 && fm.ascent() >= 11) {
+        ++extra.rheight();
+    }
+
+    setWordWrap(Qt::mightBeRichText(text()));
+
+    QSize sh = sizeHint();
+
+    // ### When the above WinRT code is fixed, windowhandle should be used to find the screen.
+    QScreen *screen = QGuiApplication::screenAt(pos);
+    if (!screen) {
+        screen = QGuiApplication::primaryScreen();
+    }
+
+    if (screen) {
+        const qreal screenWidth = screen->geometry().width();
+        if (!wordWrap() && sh.width() > screenWidth) {
+            setWordWrap(true);
+            sh = sizeHint();
+        }
+    }
+
+    resize(sh + extra);
+}
+void NotificationLabel::paintEvent(QPaintEvent *ev)
+{
+    QStylePainter p(this);
+    QStyleOptionFrame opt;
+    opt.init(this);
+    p.drawPrimitive(QStyle::PE_PanelTipLabel, opt);
+    p.end();
+    QLabel::paintEvent(ev);
+}
+void NotificationLabel::resizeEvent(QResizeEvent *e)
+{
+    QStyleHintReturnMask frameMask;
+    QStyleOption option;
+
+    option.init(this);
+
+    if (style()->styleHint(QStyle::SH_ToolTip_Mask, &option, this, &frameMask)) {
+        setMask(frameMask.region);
+    }
+
+    QLabel::resizeEvent(e);
+}
+
+NotificationLabel::~NotificationLabel()
+{
+    instance = nullptr;
+}
+void NotificationLabel::hideNotification()
+{
+    if (!hideTimer.isActive()) {
+        hideTimer.start(300);
+    }
+}
+void NotificationLabel::hideNotificationImmediately()
+{
+    close(); // to trigger QEvent::Close which stops the animation
+    deleteLater();
+}
+
+bool NotificationLabel::eventFilter(QObject *o, QEvent *e)
+{
+    Q_UNUSED(o)
+
+    switch (e->type()) {
+        case QEvent::MouseButtonPress:
+        {
+             // If minimum on screen time has already lapsed - hide the notification no matter where the click was done
+            auto total = expireTimer.interval();
+            auto remaining = expireTimer.remainingTime();
+            auto lapsed = total - remaining;
+            // ... or if the click is inside the notification, hide it no matter if the minimum onscreen time has lapsed or not
+            if( lapsed > minShowTime || this->underMouse()) {
+                hideNotification();
+                return false;
```

Comment:
It is not useless, as it hides the notification... the event is not consumed as it can be intended for elsewhere...

Reply:
My point is about 'return false;' only. It is already what the function returns globally just after. ;)

Reply:
Ah I see.

I need to check this. 

For me it is not a problem to have two "return false" here. But I think I might want to consume that event in case the click is inside the box, and not consume it if outside. I need to review this.